gnucash/report/reports/standard/income-statement.scm

```diff
@@ -295,7 +284,9 @@
   ;; get all option's values
   (let* (
 	 (report-title (get-option gnc:pagename-general optname-report-title))
-	 (company-name (get-option gnc:pagename-general optname-party-name))
+	 (book (gnc-get-current-book))
+	 ;; Get company-name should probably be modularized and standardized across reports
+	 (company-name (gnc:company-info book gnc:*company-name*))
```

Comment:
Same as on the Balance Sheet,
```
(company-name (or (gnc:company-info book gnc:*company-name*) ""))
```
It passes the tests here but without this change the report title will be "#f Income Statement for..." if there's no company name set in book options.